### PR TITLE
fix  tokenlon amount in usd

### PR DESCRIPTION
**Pull request by 0xkeen**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

**Comment by 0xkeen** [comment]
@Hosuke @jeff-dude pls review&merge

File: models/tokenlon/ethereum/tokenlon_v5_ethereum_amm_v1_trades.sql [hunk 1]
```diff
@@ -95,23 +104,23 @@ LEFT JOIN {{ ref('tokens_erc20') }} erc20a
 LEFT JOIN {{ ref('tokens_erc20') }} erc20b 
     ON erc20b.contract_address = dexs.token_sold_address
     AND erc20b.blockchain = 'ethereum'
-LEFT JOIN {{ source('prices', 'usd') }} p_bought 
-    ON p_bought.minute = date_trunc('minute', dexs.block_time)
+LEFT JOIN prices p_bought
+    ON p_bought.hour = date_trunc('hour', dexs.block_time)
```
Comment (Hosuke) [inline models/tokenlon/ethereum/tokenlon_v5_ethereum_amm_v1_trades.sql:108]: Do you have any specific reason to change `minute` into `hour`?
Like increasing precision or stability?

File: models/tokenlon/ethereum/tokenlon_v5_ethereum_amm_v1_trades.sql [hunk 2]
```diff
@@ -46,6 +46,15 @@ WITH dexs AS (
...
     WHERE evt_block_time >= date_trunc('day', now() - interval '7' day) 
     {% endif %}
+), prices AS (
+    SELECT DISTINCT
+      DATE_TRUNC('hour', minute) AS hour,
+      contract_address,
+      blockchain,
+      decimals,
+      AVG(price) AS price
+    FROM "delta_prod"."prices"."usd"
```
Comment (Hosuke) [inline models/tokenlon/ethereum/tokenlon_v5_ethereum_amm_v1_trades.sql:56]: We may not have the hardcoding source here.
```suggestion
    FROM {{ source('prices', 'usd') }}
```

File: models/tokenlon/ethereum/tokenlon_v5_ethereum_rfq_v2_trades.sql [hunk 3]
```diff
@@ -38,6 +38,15 @@ WITH dexs AS (
...
     WHERE {{ incremental_predicate('evt_block_time') }} 
     {% endif %}
+), prices AS (
+    SELECT DISTINCT
+      DATE_TRUNC('hour', minute) AS hour,
+      contract_address,
+      blockchain,
+      decimals,
+      AVG(price) AS price
+    FROM "delta_prod"."prices"."usd"
```
Comment (Hosuke) [inline models/tokenlon/ethereum/tokenlon_v5_ethereum_rfq_v2_trades.sql:48]:
```suggestion
    FROM {{ source('prices', 'usd') }}
```

File: models/tokenlon/ethereum/tokenlon_v5_ethereum_amm_v1_trades.sql (hunk 1 above)
Comment (0xkeen) [reply models/tokenlon/ethereum/tokenlon_v5_ethereum_amm_v1_trades.sql:108]: I found that using `minute`, there is missing data for certain minutes in the prices table. I can only take the average values at the `hour`

Comment (jeff-dude) [reply models/tokenlon/ethereum/tokenlon_v5_ethereum_amm_v1_trades.sql:108]: fyi @Hosuke -- this workaround relates to my latest comment in the dex issue for assigning `amount_usd`. maybe aggregating to hourly is another idea to use universally 🤔 can add to brainstorm list of ideas

Comment (Hosuke) [reply models/tokenlon/ethereum/tokenlon_v5_ethereum_amm_v1_trades.sql:108]: We may have avg-hour prices table as an independent spell.

Comment (jeff-dude) [reply models/tokenlon/ethereum/tokenlon_v5_ethereum_amm_v1_trades.sql:108]: let's approve this PR as-is in the old lineage, then continue conversation on approach in issue

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @0xkeen

**Review by jeff-dude — APPROVED** [review]
pushed a final commit to ensure jinja source syntax on all `prices.usd` reads

**Pull request merged**